### DISCUSSION TITLE
fixed nodes and connector arrows possible? (User Edition off)

### DISCUSSION THREAD

**Discussion by Klausero** [eckig/graph-editor#120] [Q&A] [answered]
Hello, I found your Library and it seems just what i desperatedly needed. I have created a javafx application and now I need to implement a Dialog with the following type of Diagram:

![imagen](https://user-images.githubusercontent.com/107101963/200933479-f034c1c7-9076-4c94-909f-01b5e8074358.png)

I would like to draw this kind of Diagram (probably creating nodes and connector arrows somewhat dinamically in a Panel), but without allowing the user to edit the components, just to visualize the nodes and their infos inside. Is it possible to draw this kind of diagram turning edition off for the user? Im looking for a library to draw this kind of Diagram.

Edit: I already downloaded the library, but it seems I cannot add the GraphEditor into a Grid panel, is it possible to add a Graph with infos inside another Panel instead of a new window?
![grafik](https://user-images.githubusercontent.com/107101963/201184461-f42272d5-46ad-488f-bed2-4a9a564155fb.png)

Thanks a lot in advance, im a bit new in this and would greatly appreciate your assistance

**Comment by eckig** [comment, marked as answer]
You need a `GraphEditorContainer` to display the Component in a JavaFX Scene Graph.

Have a look at the demo applications source code.

**Reply by Klausero** [reply]
Thank you for yor answer, I´ll have a look to the Demo regarding the container. About the proposed Diagram up there, do you think is that achievable? That side kind of Diagram with inner sub-elements and making somehow the diagram non-editable for the user?

**Reply by eckig** [reply]
With the `GraphEditorProperties` you can control what parts of the diagram are editable or disable editing alltogether.

And yes to the second question: Each diagram element would be a `GNode` and you woud assign a custom `GNodeSkin` to it. Within this custom skin, you can do whatever you would like.

So at first glance I would say, that your proposed Diagram should be entirely possible.

**Reply by Klausero** [reply]
Thanks a million Eckig, you are a life saver :)
